### PR TITLE
Add --print-ruleset option

### PR DESCRIPTION
**Pull request by Rahix**
Hi,

I was trying to build my own ruleset and didn't find any way to query the defaults.  So I thought it might make sense to add an option which prints the currently loaded ruleset.  This can then be used as a baseline to start customizing it...

What do you think?

### CODE REVIEW

File: oelint_adv/__main__.py [hunk 1]
```diff
@@ -140,6 +145,13 @@ def group_files(files):
 
     return res.values()
 
+def print_ruleset(args):
+    rules = [x for x in load_rules(args,
+        add_rules=args.addrules, add_dirs=args.customrules)]
+    ruleset = {r.ID: r.Severity for r in rules}
+    json.dump(ruleset, sys.stdout, indent=2)
```
Comment (priv-kweihmann) [inline oelint_adv/__main__.py:152]: Why not use `print(json.dumps(ruleset, indent=2, sort_keys=True) + "\n")`? I don't like to see empty print statements in the code, as they often are falsely flagged by linting tools

File: oelint_adv/__main__.py [hunk 2]
```diff
@@ -140,6 +145,13 @@ def group_files(files):
 
     return res.values()
 
+def print_ruleset(args):
+    rules = [x for x in load_rules(args,
+        add_rules=args.addrules, add_dirs=args.customrules)]
+    ruleset = {r.ID: r.Severity for r in rules}
```
Comment (priv-kweihmann) [inline oelint_adv/__main__.py:151]: As we have potentially multiple IDs per rules (and each of them can have a different severity, depending on how the __rulefile__ is configured) I would like to see that part here as a new function of `Rule` class... maybe something like `.get_active_ruleset_json()`

**Comment by priv-kweihmann** [comment]
I like the idea - please don't forget to add a test case at `tests/test_class_integration.py` to avoid future regressions

File: oelint_adv/__main__.py (hunk 1 above)
Comment (Rahix) [reply oelint_adv/__main__.py]: Right, I was just used to using the stream interface, to not allocate a large string for the json object.  But this is not that much output so it does not matter.  I changed it.

File: oelint_adv/__main__.py (hunk 2 above)
Comment (Rahix) [reply oelint_adv/__main__.py:151]: Initially, I added all the IDs, but then I noticed they weren't picked up at all - only the "main ID" did anything... Maybe I missed something?

Comment (priv-kweihmann) [reply oelint_adv/__main__.py:151]: Are you sure? For instance `oelint.var.mandatoryvar` rule should be translated to at least 4 IDs like `oelint.var.mandatoryvar.DESCRIPTION` a.s.o. - each can be controlled in severity by specifying it in the rules on its own

Comment (priv-kweihmann) [reply oelint_adv/__main__.py:151]: Huch, that doesn't look right and is likely an unrelated bug - could you please file one for that? thx

Still I think it's best to return the result after the `appendix` explode

Comment (Rahix) [reply oelint_adv/__main__.py:151]: See issue #236.

> Still I think it's best to return the result after the appendix explode

Okay, if the intended behavior was that those sub-rules are also configurable, then it makes sense.

Comment (priv-kweihmann) [reply oelint_adv/__main__.py:151]: Bug is fixed - I will wait till tomorrow (18:00 UTC) before making a new release, just to give you a bit of time to get your changes in

Comment (Rahix) [reply oelint_adv/__main__.py:151]: Found another issue with this, see #237.  The update to this PR here will now depend on the other one so you should probably merge the other one first (its commit shows up here, too).

**Comment by Rahix** [comment]
Updated the implementation to show exploded rulefile and also added a testcase.  As noted above, this PR now depends on the changes from #237.

**Comment by Rahix** [comment]
Can you retrigger CI for this PR please?  GitHub doesn't run it automatically anymore due to some abuse, IIRC.

**Comment by priv-kweihmann** [comment]
@Rahix could you please rebase and force push once more - I just fixed the issue with CI - apart from the missing checks your PR looks really good

File: oelint_adv/__main__.py [hunk 3]
```diff
@@ -62,6 +64,9 @@ def parse_arguments():
     return create_argparser().parse_args() # pragma: no cover
 
 def arguments_post(args):
+    if args.files == [] and not args.print_rulefile:
+        raise argparse.ArgumentTypeError("no input files")
```
[the remark this answers is not in the document]
Comment (Rahix) [reply oelint_adv/__main__.py:68]: Added a testcase for it.

**Comment by priv-kweihmann** [comment]
Thank you very much - will do a release soon, unless you find more issues along your way :smile:

**Comment by Rahix** [comment]
Thanks!

> will do a release soon, unless you find more issues along your way :smile:

please do, I'll stop bombarding you with PRs for now ;)  After playing with the tool a bit more (currently integrating it into a CI setup) I might have some more fixes and features for you... let's see :)

**Comment by priv-kweihmann** [comment]
Looking forward to that - happy to see this tool used in a wider scope

**Pull request merged**